src/statusbar/file-type.ts

```diff
@@ -0,0 +1,143 @@
+import { window, StatusBarAlignment, StatusBarItem } from "vscode";
+import { activeReplWindow } from '../repl-window';
+
+import { onNReplEvent } from "../nrepl";
+import { NReplEvaluationStartedEvent, NReplEvaluationFinishedEvent } from "../nrepl/events";
+
+import configReader from "../configReader";
+import * as state from '../state';
+import * as util from '../utilities';
+
+export interface EvaluationError {
+    fileName: string;
+    filePath: string;
+    message: string;
+}
+
+export class FileTypeStatusBarItem {
+    private statusBarItem: StatusBarItem;
+
+    private activeEvals = new Array<string>();
+
+    private errors = new Array<EvaluationError>();
+
+    private get isEvaluating() {
+        return this.activeEvals.length > 0;
+    }
+
+    private get hasErrors(){
+        return this.errors.length > 0;
+    }
+
+    constructor(alignment: StatusBarAlignment) {
+        this.statusBarItem = window.createStatusBarItem(alignment);
+        // TODO: Event handling and resulting state should be moved to global state
+        onNReplEvent(this.handleNreplEvent);
+    }
+
+    update() {
+        const connected = state.deref().get("connected");
+        const doc = util.getDocument({});
+        const fileType = util.getFileType(doc);
+        const sessionType = util.getREPLSessionType();
+
+        let command = null;
+        let text = "Disconnected";
+        let tooltip = "No active REPL session";
+        let color = configReader.colors.disconnected;
+
+        if(connected) {
+            if (fileType == 'cljc' && sessionType !== null && !activeReplWindow()) {
+                text = this.statusTextDecorator("cljc/" + sessionType);
+                if (util.getSession('clj') !== null && util.getSession('cljs') !== null) {
+                    command = "calva.toggleCLJCSession";
+                    tooltip = `Click to use ${(sessionType === 'clj' ? 'cljs' : 'clj')} REPL for cljc`;
+                }
+            } else if (sessionType === 'cljs') {
+                text = this.statusTextDecorator("cljs");
+                tooltip = "Connected to ClojureScript REPL";
+            } else if (sessionType === 'clj') {
+                text = this.statusTextDecorator("clj");
+                tooltip = "Connected to Clojure REPL";
+            }
+            color = this.connectedStatusColor();
+            if(this.hasErrors) {
+                tooltip = this.errorTooltip();
+            }
+        }
+
+        this.statusBarItem.command = command;
+        this.statusBarItem.text = text;
+        this.statusBarItem.tooltip = tooltip;
+        this.statusBarItem.color = color
+        this.statusBarItem.show();
+    }
+
+    private statusTextDecorator(text): string {
+        if(this.hasErrors) {
+            const c = this.errors.length;
+            text = `${text} $(alert) ${c > 1 ? c : ""}`
```

Comment:
Just curious, what's this `$(alert)` do in this template literal? Or is it just a temporary placeholder?

Reply:
It's an icon!  https://code.visualstudio.com/api/references/icons-in-labels

Reply:
Ah thanks!